android/app/src/main/java/org/jitsi/meet/AudioModeModule.java

```diff
@@ -0,0 +1,440 @@
+
+            return true;
+        }
+
+        audioManager.setMode(AudioManager.MODE_IN_COMMUNICATION);
+        audioManager.setMicrophoneMute(false);
+
+        if (audioManager.requestAudioFocus(null,
+                AudioManager.STREAM_VOICE_CALL, AudioManager.AUDIOFOCUS_GAIN) ==
+                AudioManager.AUDIOFOCUS_REQUEST_FAILED) {
+                Log.d(TAG, "Audio focus request failed");
+                return false;
+        }
+
+        boolean useSpeaker = (mode == VIDEO_CALL);
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            // On Android >= M we use the AudioDeviceCallback API, so turn on Bluetooth SCO
+            // from the start.
+            if (audioManager.isBluetoothScoAvailableOffCall()) {
+                audioManager.startBluetoothSco();
+            }
+        } else {
+            // On older Android versions we must set the Bluetooth route manually. Also disable
+            // the speaker in that case.
+            setBluetoothAudioRoute(bluetoothManager.isHeadsetAvailable());
+            if (bluetoothManager.isHeadsetAvailable()) {
+                useSpeaker = false;
+            }
+        }
+
+        audioManager.setSpeakerphoneOn(useSpeaker &&
+                !(audioManager.isWiredHeadsetOn() || audioManager.isBluetoothScoOn()));
+
+        return true;
+    }
+
+    /**
+     * Public method to set the current audio mode.
+     *
+     * @param mode the desired audio mode.
+     * @param promise a {@link Promise} which will be resolved if the audio mode could be updated
+     *                successfully, and it will be rejected otherwise.
+     */
+    @ReactMethod
+    public void setMode(final int mode, final Promise promise) {
+        Runnable r = new Runnable() {
+            @Override
+            public void run() {
+                if (updateAudioRoute(mode)) {
+                    currentMode = mode;
+                    promise.resolve(null);
+                } else {
+                    promise.reject("setMode", "Failed to set the requested audio mode");
+                }
+            }
+        };
+        mainThreadHandler.post(r);
+    }
+
+    /**
+     * Setup the audio route change detection mechanism. We use the
+     * {@link android.media.AudioDeviceCallback} API on Android >= 23 only.
+     */
+    private void setupAudioRouteChangeDetection() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            setupAudioRouteChangeDetectionM();
+        } else {
+            setupAudioRouteChangeDetectionOld();
+        }
+    }
+
+    /**
+     * Audio route change detection mechanism for Android API >= 23.
+     */
+    @TargetApi(Build.VERSION_CODES.M)
+    private void setupAudioRouteChangeDetectionM() {
+        android.media.AudioDeviceCallback audioDeviceCallback =
+                new android.media.AudioDeviceCallback() {
+                    public void onAudioDevicesAdded(AudioDeviceInfo[] addedDevices) {
```

Comment:
I'm guessing these are interface implementations. I think there're other cases in the PR where you added `@Override` so why not here?